terraform/newsletter_lambda.tf

```diff
@@ -0,0 +1,116 @@
+
+# lambda repo clone. 
+resource "null_resource" "aws_lambda_repo_clone" {
+  provisioner "local-exec" {
+    command     = <<-EOT
+    git clone ${var.github_repo}
+    pip3 install -r ${var.temp_path}/${var.foldername}/${var.github_repo_name}/requirements.txt -t ${var.temp_path}/${var.foldername}/${var.github_repo_name}/
```

Comment:
We can also probable add a check for 'pip' of some sort. Some users may have `pip`, some may have `pip3`. We need to handle this scenario

Reply:
# Determine the correct pip executable (pip3 or pip) and install dependencies
$(command -v pip3 >/dev/null && echo "pip3" || echo "pip") install -r ${var.temp_path}/${var.foldername}/${var.github_repo_name}/requirements.txt -t ${var.temp_path}/${var.foldername}/${var.github_repo_name}/